tests/mps_statespace_test.cc

```diff
@@ -75,11 +75,304 @@ TEST(MPSStateSpaceTest, Copy) {
   }
 }
 
+TEST(MPSStateSpaceTest, ToWaveFunctionZero) {
+  auto ss = MPSStateSpace<For, float>(1);
+  auto mps = ss.CreateMPS(2, 8);
+  ss.SetMPSZero(mps);
+  float *wf = new float[8];
+  ss.ToWaveFunction(mps, wf);
+  EXPECT_NEAR(wf[0], 1, 1e-5);
+  for (int i = 1; i < 8; i++) {
+    EXPECT_NEAR(wf[i], 0, 1e-5);
+  }
+  delete[](wf);
+}
+
+TEST(MPSStateSpaceTest, ToWaveFunction3) {
+  auto ss = MPSStateSpace<For, float>(1);
+  auto mps = ss.CreateMPS(3, 4);
+
+  // Set to highly entangled three qubit state.
```

Comment:
What was used to contruct these MPS states and wavefunctions? If an error occurred in this test, I wouldn't know what the issue was outside of "the MPS simulator is broken."

Reply:
It's just an arbitrary wavefunction I made using numpy. From there I worked backwards to MPS matrix entries. You could also go the other way and produce some random matrix entries and then infer what the state vector ought to be. Do you have any suggestions on improving these tests ?

Reply:
I'd be more confident in a test that verified this against some other MPS-to-wavefunction conversion, or provided comments detailing how to do that conversion. In the end, though, this is probably more my own lack of familiarity showing rather than a true issue with the test - I'll leave it up to you whether to clarify further.

Reply:
I personally think this isolates the behavior of the wavefunction conversion reasonably well. If we relied on some other MPS to wavefunction converter to check our own MPS to wavefunction converter it kind of feels like the unit test wouldn't be very targeted and would be more like some kind of fuzz test. I'll go ahead and add a comment explaining the desired transformation we want the MPS blocks to undergo for each test.

Reply:
Thanks, the comment makes this much clearer!